src/concrete/ml/onnx/ops_impl.py

```diff
@@ -897,7 +899,43 @@ def numpy_equal(
     Returns:
         Tuple[numpy.ndarray]: Output tensor
     """
+    return (numpy.equal(x, y),)
+
+
+def rounded_numpy_equal_for_trees(
+    x: numpy.ndarray,
+    y: numpy.ndarray,
+    *,
+    lsbs_to_remove_for_trees: Optional[int] = None,
+) -> Tuple[numpy.ndarray]:
+    """Compute rounded equal in numpy according to ONNX spec for tree-based models only.
 
+    See https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Equal-11
+
+    Args:
+        x (numpy.ndarray): Input tensor
+        y (numpy.ndarray): Input tensor
+        lsbs_to_remove_for_trees (Optional[int]): Number of the least significant bits to remove
+            for tree-based models only.
+
+    Returns:
+        Tuple[numpy.ndarray]: Output tensor
+    """
+
+    # For tree-based models in the second stage, x == y is equivalent to x <= y or x < y - 1
```

Comment:
I am not sure I understand what we do here. `x <= y or x < y - 1` where do we apply this?

Reply:
We had a discussion on slack regarding this: https://zama-ai.slack.com/archives/C0455QDGT6C/p1700036915697839

Reply:
@jfrery, this part will disappear with rounding

Reply:
Alright